docs/core/diagnostics/snippets/OTel-Prometheus-Grafana-Yaeger/csharp/Program.cs

```diff
@@ -0,0 +1,158 @@
+﻿// <WholeProgram>
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Logs;
+using Azure.Monitor.OpenTelemetry.AspNetCore;
+using Azure;
+
+var builder = WebApplication.CreateBuilder(args);
+
+
+// <Snippet_CustomMetrics>
+// Custom metrics for the application
+var GreeterMeter = new Meter("GOtPrGrYa.Sample", "1.0.0");
+var CountGreetings = GreeterMeter.CreateCounter<int>("greetings.count", description: "Counts the number of greetings");
+
+// Custom ActivitySource for the application
+var GreeterActivitySource = new ActivitySource("OtPrGrYa.Sample");
+// </Snippet_CustomMetrics>
+// <Snippet_HttpClientFactory>
+builder.Services.AddHttpClient();
+// </Snippet_HttpClientFactory>
+
+
+#if !AZURE_MONITOR
+// <Snippet_OTEL>
+var TracingOtlpEndpoint = builder.Configuration["OTLP_ENDPOINT_URL"];
```

Comment:
Should be lowercase

Reply:
@davidfowl - I've seen you respond this way before to all cap config keys, I'm curious why? Is that a standard, or something that I'm not familiar with? It's very common in practice, and the underlying config store is case insensitive (right) - what am I missing? Lots of devs tend to use the same key name and same casing when using environment variables, most often all caps.

Reply:
I meant variables (like TracingOtlpEndpoint), not the configuration key. My bad. Also, configuration keys are not all uppercase unless they are specifically environment variables.